src/core/utils/urlutils.js

```diff
@@ -82,3 +84,66 @@ export function equivalentParams (params1, params2) {
   }
   return true;
 }
+
+/**
+ * Creates a new params object without params that begin with the given prefixes
+ * @param {SearchParams} params The parameters to remove from
+ * @param {string[]} prefixes The prefixes of parameters to remove
+ * @return {SearchParams} A new instance of SearchParams without removed params
+ *   from the params parameter
+ */
+export function removeParamsWithPrefixes (params, prefixes) {
+  const newParams = new SearchParams();
+  for (const [key, val] of params.entries()) {
+    let includeEntry = true;
+    for (const prefix of prefixes) {
```

Comment:
Minor nit, but we can streamline this a bit:
includeEntry = prefixes.some(prefix => key.startsWith(prefix));
Whenever possible, we try and do things in the more functional style (a holdover from our Java days).